widget/embedded/src/pages/ConfirmSwapPage.tsx

```diff
@@ -224,19 +231,41 @@ export function ConfirmSwapPage(props: PropTypes) {
         )
       );
 
-    const routeWalletsChanged = confirmSwapDisabled(
-      fetchingConfirmationRoute,
-      !!customDestination,
-      customDestination,
-      bestRoute,
-      selectedWallets,
-      destination
+    let routeWalletsChanged = false;
+
+    let requiredWallets = getRequiredWallets(bestRoute);
+
+    const lastStepToBlockchain =
+      bestRoute?.result?.swaps[bestRoute.result.swaps.length - 1].to.blockchain;
+
+    const isLastWalletRequired = !!bestRoute?.result?.swaps.find(
+      (swap) => swap.from.blockchain === lastStepToBlockchain
     );
 
+    if (!isLastWalletRequired) {
+      requiredWallets = requiredWallets.slice(
+        -requiredWallets.length,
```

Comment:
Do we need negative index here. I'm not sure.

Reply:
yes